### ISSUE TITLE
sort-comp with type-annotations does not work if the property is initialized with a value

### ISSUE THREAD

**Issue by akkie** [yannickcr/eslint-plugin-react#1108]
The following code shows the warning: initialized should be placed after constructor

```js
class Test extends React.Component {
  initialized: boolean = true;
  constructor(props) {
    super(props);
  }
}
```

And if I remove the value then it works as expected:
```js
class Test extends React.Component {
  initialized: boolean;
  constructor(props) {
    super(props);
  }
}
```

**Comment by ljharb** [comment]
What order does it want if you do `initialized = true;` (ie, if you remove the type annotation)?

I suspect that the "type annotations" ordering isn't considering initializers as being type annotated - since `initialized: boolean` is flow-*only*, but `initialized: boolean = true` is a normal thing that's flow-annotated.

**Comment by akkie** [comment]
@ljharb If I remove the type annotation, then I get the same warning message: initialized should be placed after constructor

**Comment by ljharb** [comment]
Right - because "initializes a value" takes precedence over "type annotation". This seems correct to me.

If you want the type annotation up top, I think you should use two statements - `initialized: boolean` and then `initialized = true`.

**Comment by ljharb** [comment]
I believe the "instance variables" and "types" groups should handle this now.
